packages/yarnpkg-core/sources/StreamReport.ts

```diff
@@ -170,7 +174,7 @@ export class StreamReport extends Report {
   private progressMaxScaledSize: number;
 
   private forgettableBufferSize: number;
-  private forgettableNames: Set<MessageName | null>;
+  private forgettableNames: Set<MessageName | PluginMessageName | null>;
```

Comment:
Rather than use `MessageName | PluginMessageName` everywhere, what about renaming `MessageName` into `CoreMessageName` , and aliasing `MessageName = CoreMessageName | PluginMessageName`?